src/flow_solver/flow_solver_cases/periodic_cube_flow.cpp

```diff
@@ -22,12 +33,35 @@ PeriodicCubeFlow<dim, nstate>::PeriodicCubeFlow(const PHiLiP::Parameters::AllPar
 template <int dim, int nstate>
 std::shared_ptr<Triangulation> PeriodicCubeFlow<dim,nstate>::generate_grid() const
 {
+
+    const int mpi_rank = dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+    dealii::ConditionalOStream pcout(std::cout, mpi_rank==0);
+
     std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (
 #if PHILIP_DIM!=1
             this->mpi_communicator
 #endif
     );
-    Grids::straight_periodic_cube<dim,Triangulation>(grid, domain_left, domain_right, number_of_cells_per_direction);
+
+    if (!this->all_param.flow_solver_param.use_input_mesh) {
+
+        pcout << "Using Internal Deal.ii HyperCube Generator." << std::endl;
+
+        Grids::straight_periodic_cube<dim, Triangulation>(grid, domain_left, domain_right,
+                                                          number_of_cells_per_direction);
+
+    } else {
+        if constexpr(dim == 3)
```

Comment:
what if `dim!=3`?

Reply:
Not tested yet for dim != 3. Most likely need a 2D case. Current working prototype is for the Cube TGV case,